handler.py

```diff
@@ -1,54 +1,37 @@
 import json
 import datetime
 import requests
-#from SimpleHTTPServer import SimpleHTTPRequestHandler
-#import SocketHandler
-RASASERVER_URL = "http://ec2-34-218-219-244.us-west-2.compute.amazonaws.com:5000/parse"
-
-def endpoint(event, context):
-    current_time = datetime.datetime.now().time()
-
-    if str(event['requestContext']['httpMethod']) == "GET":
-        body = {
-            "request type": "" + str(event)
-        }
-    elif str(event['requestContext']['httpMethod']) == "POST":
 
-        wrappedPayload = json.loads(event['body'])
-        print("wrappedPayload: ", wrappedPayload)
-        payload = wrappedPayload['payload']
-        r = requests.post(RASASERVER_URL, json={"q": payload})
-        response = r.json()
-        intentString = response['intent']['name']
-        responsePayload = {
-            "actionType": "speak",
-            "actionDetail": intentString
-        }
-        body = {
-            "receiver": wrappedPayload["sender"],
-            "payload": responsePayload
-        }
+RASASERVER_URL = "http://ec2-34-218-219-244.us-west-2.compute.amazonaws.com:5000/parse"
 
+def isPostRequest(event):
+    return str(event['requestContext']['httpMethod']) == "POST"
 
-    else:
-        body={
-            "request type": "Neither GET or POST"
-        }
+def unwrapEvent(event):
+    wrappedPayload = json.loads(event['body'])
+    return wrappedPayload['payload'], wrappedPayload['sender']
 
+def postRasaForIntent(payload):
+    r = requests.post(RASASERVER_URL, json={"q": payload})
+    return r.json()
 
-    #body = {
-    #    "heres a header!"
-    #}
-    #else:
-    #    body = "request type": "" + str(event['requestContext']['httpMethod'])
+def intentNameFromRasaJson(json):
+    return rasaResponse['intent']['name']
 
-    response = {
-        "statusCode": 200,
-        "body": json.dumps(body)
+def wrapIntentSpeakAction(json, sender):
```

Comment:
can't use json as a variable since you are using a method from the json import below